error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/OptionalRules.java

```diff
@@ -355,6 +355,42 @@ Optional<T> after(Optional<T> optional) {
     }
   }
 
+  /**
+   * Avoid unnecessary {@link Optional} to {@link Stream} conversion when filtering a value of the
+   * former type.
+   */
+  static final class OptionalFilter<T> {
+    @BeforeTemplate
+    Optional<T> before(Optional<T> optional, Predicate<? super T> predicate) {
+      return Refaster.anyOf(
+          optional.stream().filter(predicate).findFirst(),
+          optional.stream().filter(predicate).findAny());
+    }
+
+    @AfterTemplate
+    Optional<T> after(Optional<T> optional, Predicate<? super T> predicate) {
+      return optional.filter(predicate);
+    }
+  }
+
+  /**
+   * Avoid unnecessary {@link Optional} to {@link Stream} conversion when mapping a value of the
+   * former type.
+   */
+  // XXX: If `StreamMapFirst` also simplifies `.findAny()` expressions, then this rule can be
```

Comment:
IIUC in that case the `self-check` would identify this case right? 

In that case I'd argue that it would be better to drop this XXX 😬. 
@Stephan202

Reply:
I agree that we _could_ drop it for that reason, but without it on might ask "why not add `.findFirst()`?" IMHO now it's clearer why this rule exists in the first place :shrug:

Reply:
Okay okay. Agreed, it doesn't really fit in a `@Description` or something like that, so XXX is most reasonable.